tooling/greybox_fuzzer/src/mutation/configurations.rs

```diff
@@ -0,0 +1,874 @@
+        selector -= self.subtraction_weight;
+        if selector < self.and_operation_weight {
+            return BinaryIntOperationMutation::And;
+        }
+        selector -= self.and_operation_weight;
+        if selector < self.or_operation_weight {
+            return BinaryIntOperationMutation::Or;
+        }
+        BinaryIntOperationMutation::Xor
+    }
+}
+pub(crate) enum IntTopLevelMutation {
+    FixedSubstitution,
+    DictionarySubstitution,
+    Negation,
+    Shift,
+    SmallValueUpdate,
+    DictionaryValueUpdate,
+}
+
+pub(crate) struct IntTopLevelMutationConfiguration {
+    fixed_substitution_weight: usize,
+    dictionary_substitution_weight: usize,
+    negation_weight: usize,
+    shift_weight: usize,
+    small_value_update_weight: usize,
+    #[allow(unused)]
+    dictionary_value_update_weight: usize,
+    total_weight: usize,
+}
+
+impl IntTopLevelMutationConfiguration {
+    #[allow(unused)]
+    pub fn new(
+        fixed_substitution_weight: usize,
+        dictionary_substitution_weight: usize,
+        negation_weight: usize,
+        shift_weight: usize,
+        small_value_update_weight: usize,
+        dictionary_value_update_weight: usize,
+    ) -> Self {
+        let total_weight = fixed_substitution_weight
+            + dictionary_substitution_weight
+            + negation_weight
+            + shift_weight
+            + small_value_update_weight
+            + dictionary_value_update_weight;
+        Self {
+            fixed_substitution_weight,
+            dictionary_substitution_weight,
+            negation_weight,
+            shift_weight,
+            small_value_update_weight,
+            dictionary_value_update_weight,
+            total_weight,
+        }
+    }
+
+    /// Select a mutation according to weights
+    pub fn select(&self, prng: &mut XorShiftRng) -> IntTopLevelMutation {
+        let mut selector = prng.gen_range(0..self.total_weight);
+        if selector < self.fixed_substitution_weight {
+            return IntTopLevelMutation::FixedSubstitution;
+        }
+        selector -= self.fixed_substitution_weight;
+        if selector < self.dictionary_substitution_weight {
+            return IntTopLevelMutation::DictionarySubstitution;
+        }
+        selector -= self.dictionary_substitution_weight;
+        if selector < self.negation_weight {
+            return IntTopLevelMutation::Negation;
+        }
+        selector -= self.negation_weight;
+        if selector < self.shift_weight {
+            return IntTopLevelMutation::Shift;
+        }
+        selector -= self.shift_weight;
+        if selector < self.small_value_update_weight {
+            return IntTopLevelMutation::SmallValueUpdate;
+        }
+        IntTopLevelMutation::DictionaryValueUpdate
+    }
+}
+
+/// Default configurations for all mutations that are currently used
+
+pub(crate) const BASIC_SPLICE_MUTATION_CONFIGURATION: SpliceMutationConfiguration =
+    SpliceMutationConfiguration {
```

Comment:
Would it make sense to use `::new` here? It requires `new` to be `const` but it works, and then we don't have to manually compute `total_weight` (though I don't know if the reason for now doing that is making it more clear what the numbers mean)

Reply:
Actually, nevermind. After seeing a few of these I understand now that the unused value is the last case that will be considered, and it's better to have explicit names here (in other cases it's more than just two possibilities).